crates/batch-prover/src/proving.rs

```diff
@@ -221,8 +222,18 @@ where
         }
     }
 
+    let last_l2_height = sequencer_commitments
+        .last()
+        .expect("Should have at least 1 commitment")
+        .l2_end_block_number;
+    let current_spec = fork_from_block_number(FORKS, last_l2_height).spec_id;
+    let elf = elfs_by_spec
+        .get(&current_spec)
+        .expect("Every fork should have an elf attached")
+        .clone();
+
     // Prove all proofs in parallel
-    let proofs = prover_service.prove().await?;
+    let proofs = prover_service.prove(elf).await?;
```

Comment:
this is good, however, we'll need to generate V1 input if we prove for older elfs on testnet or mainnet

Reply:
Will be done in: https://github.com/chainwayxyz/citrea/issues/1538